### PR TITLE
Port over members page

### LINKED ISSUE #40: Port over Members page
This includes:
- components, which should live in `components/common`
- update and introduce new route
- decide on how to fetch the members, watches, reputation (from the db/lambda fn obviously)
- add required queries

Note that this should be split up into multiple issues

### PR DESCRIPTION
**Pull request by danbr**
## Description

This is the feature branch porting over the Member's page.

- [x] Create lambda function to query members
- [x] Update Member's aside on Colony home to support  lambda

Todo:
- [x] Update Member's page to support  lambda #101

Resolves #40

### CODE REVIEW

**Review by jakubcolony — COMMENTED** [review]
Nice! The code looks good, unfortunately I'm still unable to test this PR as I'm getting the following error from lambda functions:
>"Colony version not supported"

@rdig mentioned previously it's a matter of changing network image hash from `develop` to `e2194001e88dc129213a626a7d38943cbb0c9ef1` but that doesn't seem to help. I also tried installing a different version of `@colony/colony-js` but had no luck.

**Comment by Julianahlk** [comment]
> Nice! The code looks good, unfortunately I'm still unable to test this PR as I'm getting the following error from lambda functions:
> 
> > "Colony version not supported"

@jakubcolony did you restart the dev environment after switching to this branch?

File: src/graphql/fragments/user.graphql
Comment (Julianahlk) [reply src/graphql/fragments/user.graphql:35]: @jakubcolony if the User fragment is used, wouldn't that also include watchlist which is unnecessary?

Comment (jakubcolony) [reply src/graphql/fragments/user.graphql:35]: Sorry for the confusion, I didn't necessarily mean the existing "User" fragment but rather a fragment on User or Profile type that would extract the duplicated fields

**Comment by jakubcolony** [comment]
>@jakubcolony did you restart the dev environment after switching to this branch?

Yes, many times - unfortunately without luck

File: src/components/common/ColonyHome/ColonyMembersWidget/ColonyMembersWidget.tsx
Comment (jakubcolony) [inline src/components/common/ColonyHome/ColonyMembersWidget/ColonyMembersWidget.tsx:9]: Styles should go below

File: src/components/common/ColonyHome/ColonyMembersWidget/MembersSubsection.tsx
Comment (jakubcolony) [inline src/components/common/ColonyHome/ColonyMembersWidget/MembersSubsection.tsx:62]: Does it still make sense for it to be optional? You seem to provide a fallback of empty array in `ColonyMembersWidget`

File: src/components/common/ColonyHome/ColonyMembersWidget/MembersSubsection.tsx
Comment (jakubcolony) [inline src/components/common/ColonyHome/ColonyMembersWidget/MembersSubsection.tsx:129]: For the reason I mentioned in my previous comment (the fallback for contributors/watchers), I don't think this will ever be true

**Review by jakubcolony — CHANGES_REQUESTED** [review]
Got it to work after all! Looking good, just left a couple of comments

File: src/components/common/ColonyHome/ColonyMembersWidget/MembersSubsection.tsx
Comment (Julianahlk) [reply src/components/common/ColonyHome/ColonyMembersWidget/MembersSubsection.tsx:129]: Yes, makes sense, updating to `!members.length` to check for an empty array instead

**Comment by Julianahlk** [comment]
> Got it to work after all! Looking good, just left a couple of comments

@jakubcolony Thank you for your comments. I already made the changes you suggested, please take a look again whenever you can.

**Review by willm30 — COMMENTED** [review]
Is this being used as the feature branch? If so, I will wait until the project is complete to review.  Perhaps move back into draft as well for time being so we know it's covering the entire feature.

File: src/components/common/ColonyHome/ColonyMembersWidget/MembersSubsection.tsx
Comment (jakubcolony) [inline src/components/common/ColonyHome/ColonyMembersWidget/MembersSubsection.tsx:129]: This will be true if the members array is empty, which is expected to happen and is not necessarily an error. We shouldn't be showing the error message here. 

Can I suggest that you pass the `members` prop with its original type (including null and undefined) and if it's null or undefined, then show the error here. If it's just empty, then we shouldn't show anything

File: src/components/common/ColonyHome/ColonyMembersWidget/MembersSubsection.tsx
Comment (jakubcolony) [inline src/components/common/ColonyHome/ColonyMembersWidget/MembersSubsection.tsx:155]: It doesn't look like `member` can currently be undefined so could do without the optional chaining 
If you do go with my suggestion above, then you can run `filter(notNull)` on the `members` array to remove null items

File: src/graphql/fragments/user.graphql
Comment (jakubcolony) [inline src/graphql/fragments/user.graphql:47]: Nice and clean!

**Review by jakubcolony — CHANGES_REQUESTED** [review]
Great stuff, thanks for implementing the changes we talked about!
I left some comments regarding how we handle colonies without contributors - but all good otherwise! 👍

File: src/components/common/ColonyHome/ColonyMembersWidget/MembersSubsection.tsx
Comment (Julianahlk) [reply src/components/common/ColonyHome/ColonyMembersWidget/MembersSubsection.tsx:83]: no, It should always atleast be an empty array

File: src/components/common/ColonyHome/ColonyMembersWidget/MembersSubsection.tsx
Comment (Julianahlk) [reply src/components/common/ColonyHome/ColonyMembersWidget/MembersSubsection.tsx:155]: Nice, didn't know about notNull ~utils/arrays

**Comment by Julianahlk** [comment]
> Is this being used as the feature branch? If so, I will wait until the project is complete to review. Perhaps move back into draft as well for time being so we know it's covering the entire feature.

@willm30 yes this is the feature branch, I wanted to have some eyes on it first but we can wait until #101 is merged to complete the review.

**Review by rdig — APPROVED** [review]
All good for me, especially after you addressed Jakub's changes.

I'll prioritize reviewing #101 as well so you can merge as well

File: src/components/common/ColonyHome/ColonyMembersWidget/MembersSubsection.tsx
Comment (Julianahlk) [reply src/components/common/ColonyHome/ColonyMembersWidget/MembersSubsection.tsx]: Thanks @jakubcolony for all your comments and suggestions. They are all resolved now 😉.

File: src/components/common/ColonyHome/ColonyMembersWidget/MembersSubsection.tsx
Comment (jakubcolony) [inline src/components/common/ColonyHome/ColonyMembersWidget/MembersSubsection.tsx:16]: Should be above styles

File: src/components/common/ColonyHome/ColonyMembersWidget/MembersSubsection.tsx
Comment (jakubcolony) [inline src/components/common/ColonyHome/ColonyMembersWidget/MembersSubsection.tsx:130]: Why the non-strict equality check instead of `!members`?

**Review by jakubcolony — APPROVED** [review]
Great work, all looks & works good! 👍

**Comment by Julianahlk** [comment]
@rdig @jakubcolony All pending branches have been merged into this members feature branch. Do you want to re-review before this gets merged?

**Comment by rdig** [comment]
Are we handling the filtering/sorting of members in this branch as well ? If not, we can merge

**Comment by Julianahlk** [comment]
> Are we handling the filtering/sorting of members _(using the lambda function)_ in this branch as well ? If not, we can merge

@rdig Issue #276 was opened to handle filtering/sorting and hasn't been started yet.

**Comment by rdig** [comment]
> > Are we handling the filtering/sorting of members _(using the lambda function)_ in this branch as well ? If not, we can merge
> 
> @rdig Issue #276 was opened to handle filtering/sorting and hasn't been started yet.

And do you want to merge it in here (once it's done), or are you happy merging it directly into mater ?

**Comment by Julianahlk** [comment]
> I notice we're still using formik in a few places. Should we add #157 to this branch since it's the feature branch?

@willm30 right, makes sense all working on get that completed and merged here first

**Pull request merged**
